github/github.go

```diff
@@ -101,6 +101,13 @@ func (c *Client) ParseFromURL(baseRepoURL, repoName string) (RepoInfo, error) {
 		return ret, fmt.Errorf("error creating request: %w", err)
 	}
 
+	// authenticate the request if there is a token
+	// this will lower the change of hitting the rate limit
+	auth, present := os.LookupEnv("GITHUB_AUTH_TOKEN")
+	if present {
+		req.SetBasicAuth("x", auth)
+	}
+
```

Comment:
`Client` already uses the roundtripper from scorecard that will read this environment variable. 
What do you think about changing the call below to use a http.Client that uses that roundtripper?